extensions/resteasy-reactive/quarkus-rest-client-microprofile/deployment/src/main/java/io/quarkus/resteasy/reactive/client/deployment/ReactiveResteasyMpClientProcessor.java

```diff
@@ -114,13 +117,16 @@ void addRestClientBeans(Capabilities capabilities,
                             configPrefix);
                     classCreator.addAnnotation(scope.getDotName().toString());
                     classCreator.addAnnotation(RestClient.class);
+                    org.objectweb.asm.Type asmType = org.objectweb.asm.Type
+                            .getObjectType(jaxrsInterface.name().toString().replace('.', '/'));
+                    classCreator.addAnnotation(Typed.class.getName(), RetentionPolicy.RUNTIME)
```

Comment:
Me and Michal both tried to temper with this to replace it with just pure Jandex (instead of asm) but failed.
The closest we got was with something like this (but that doesn't work either):
```
classCreator.addAnnotation(AnnotationInstance.create(TYPED, jaxrsInterface,
                            new AnnotationValue[] { AnnotationValue.createArrayValue("value",
                                    new AnnotationValue[] {
                                            AnnotationValue.createClassValue("value",
                                                    Type.create(jaxrsInterface.name(), org.jboss.jandex.Type.Kind.CLASS))
                                    })
                            }));
```

But maybe we're missing something. @n1hility is the above not a proper way to declare an annotation value that's an array of classes?